src/misc/ethernet_pcap.cpp

```diff
@@ -0,0 +1,225 @@
+                niclist+=(i<10?"0":"")+std::to_string(i)+" "+currentdev->name+"\n    ("+desc+")\n";
+            }
+        }
+	if (!strcasecmp(realnicstring,"list")) {
+			// print list and quit
+            std::istringstream in(("\n"+niclist+"\n").c_str());
+            if (in)	for (std::string line; std::getline(in, line); )
+                LOG_MSG("%s", line.c_str());
+			pcap_freealldevs(alldevs);
+			return false;
+	} else if(1==sscanf(realnicstring,"%u",&userdev)) {
+		// user passed us a number
+		Bitu i = 0;
+		currentdev=alldevs;
+		while(currentdev!=NULL) {
+			i++;
+			if(i==userdev) break;
+			else currentdev=currentdev->next;
+		}
+	} else {
+		// user might have passed a piece of name
+		for(currentdev=alldevs; currentdev!=NULL; currentdev=currentdev->next) {
+			if(strstr(currentdev->name,realnicstring)) {
+				break;
+			}else if(currentdev->description!=NULL &&
+				strstr(currentdev->description,realnicstring)) {
+				break;
+			}
+		}
+	}
+
+	if(currentdev==NULL) {
+		LOG_MSG("Unable to find network interface - check realnic parameter\n");
+		pcap_freealldevs(alldevs);
+		return false;
+	}
+	// print out which interface we are going to use
+        const char* desc = "no description"; 
+	if(currentdev->description) desc=currentdev->description;
+	LOG_MSG("Using Network interface:\n%s\n(%s)\n",currentdev->name,desc);
+	
+	// attempt to open it
+#ifdef WIN32
+	if ( (adhandle= pcap_open(
+			currentdev->name, // name of the device
+            65536,            // portion of the packet to capture
+                              // 65536 = whole packet 
+            PCAP_OPENFLAG_PROMISCUOUS,    // promiscuous mode
+            -1,             // read timeout
+            NULL,             // authentication on the remote machine
+            errbuf            // error buffer
+            ) ) == NULL)
+#else
+	/*pcap_t *pcap_open_live(const char *device, int snaplen,
+        int promisc, int to_ms, char *errbuf)*/
+	if ( (adhandle= pcap_open_live(
+			currentdev->name, // name of the device
+            65536,            // portion of the packet to capture
+                              // 65536 = whole packet 
+            true,    // promiscuous mode
+            -1,             // read timeout
+            errbuf            // error buffer
+            ) ) == NULL)
+
+#endif        
+        {
+		LOG_MSG("\nUnable to open the interface: %s.", errbuf);
+        	pcap_freealldevs(alldevs);
+		return false;
+	}
+	pcap_freealldevs(alldevs);
+#ifndef WIN32
+	pcap_setnonblock(adhandle,1,errbuf);
+#endif
+	return true;
+}
+
+void PcapEthernetConnection::SendPacket(uint8_t* packet, int len)
+{
+	pcap_sendpacket(adhandle, packet, len);
+}
+
+void PcapEthernetConnection::GetPackets(std::function<void(uint8_t*, int)> callback)
+{
+	int res;
+	struct pcap_pkthdr *header;
+	u_char *pkt_data;
+//#if 0
+	while((res = pcap_next_ex( adhandle, &header, (const u_char **)&pkt_data)) > 0) {
+		callback(pkt_data, header->len);
+	}
+//#endif
```

Comment:
Suggest similar feedback of number of packets received (tallied in the while-loop) or `-1` on error, with similar error-message logging.

> pcap_next_ex() returns 1 if the packet was read without problems, 0 if packets are being read from a live capture and the packet buffer timeout expired, PCAP_ERROR if an error occurred while reading the packet, and PCAP_ERROR_BREAK if packets are being read from a ``savefile'' and there are no more packets to read from the savefile. If PCAP_ERROR is returned, pcap_geterr(3PCAP) or pcap_perror(3PCAP) may be called with p as an argument to fetch or display the error text. 

https://www.tcpdump.org/manpages/pcap_next_ex.3pcap.html